src/dune_rules/fetch_rules.ml

```diff
@@ -206,6 +206,7 @@ let gen_rules_for_checksum_or_url (loc_url, (url : OpamUrl.t)) checksum =
     let action ~target ~kind =
       action ~url:(loc_url, url) ~checksum ~target ~kind
       |> Action.Full.make
```

Comment:
Actually, you can just pass the `can_go_in_shared_cache` argument to `Action.Full.make`.

Reply:
True, that's a better solution. Completely missed that.